src/main/java/org/opensearch/security/dlic/rest/validation/RequestContentValidator.java

```diff
@@ -247,10 +247,8 @@ private ValidationResult<JsonNode> validatePassword(final RestRequest request, f
                 this.validationError = ValidationError.INVALID_PASSWORD_TOO_SHORT;
                 return ValidationResult.error(RestStatus.BAD_REQUEST, this);
             }
-            final String username = Utils.coalesce(
-                request.param("name"),
-                validationContext.hasParams() ? (String) validationContext.params()[0] : null
-            );
+            final String username = Optional.ofNullable(request.param("name"))
+                .orElseGet(() -> validationContext.hasParams() ? (String) validationContext.params()[0] : null);
```

Comment:
This codes doesn't make sense to me, do you know what this code is suppose to be doing? Where is the username coming from if it isn't part of the parameters on the request

Reply:
Not nice solution agree. But this is how we validate passwords in case of PATCH since we have username in the path. I'm sure better solution exists but so far it is what it is.